tools/functional-tester/etcd-runner/command/election_command.go

```diff
@@ -65,18 +71,30 @@ func runElectionFunc(cmd *cobra.Command, args []string) {
 				break
 			}
 		}
-		e := concurrency.NewElection(s, "electors")
 
-		rcs[i].acquire = func() error {
-			<-releasec
+		e := concurrency.NewElection(s, election)
+		rcs[i].acquire = func() (err error) {
 			ctx, cancel := context.WithCancel(context.Background())
+			donec := make(chan struct{})
 			go func() {
-				if ol, ok := <-e.Observe(ctx); ok {
+				defer close(donec)
+				ol, ok := <-e.Observe(ctx)
```

Comment:
does this path need to call `Leader()`? it should probably call observe again instead of spinning on `Leader()`:
```go
defer close(donec)
for ctx.Err() == nil {
    if ol, ok := <-e.Observe(ctx); ok {
        observedLeader = string(ol.Kvs[0].Value)
        break
    }
}
if observedLeader != v {
    cancel()
}
```